internal/backup_push_handler.go

```diff
@@ -232,3 +233,10 @@ func UploadSentinel(uploader *Uploader, sentinelDto *BackupSentinelDto, backupNa
 
 	return uploader.Upload(sentinelName, bytes.NewReader(dtoBody))
 }
+
+// This function is needed for being cross-platform
+func ceilTimeUpToMicroseconds(timeToCeil time.Time) time.Time {
```

Comment:
1. Let's unit test it
2. It doesn't work correctly when timeToCeil.Nanosecond() % 1000 == 0